presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewInformationExtractor.java

```diff
@@ -173,5 +168,26 @@ public boolean isDistinct()
         {
             return isDistinct;
         }
+
+        public boolean hasGroupBy()
+        {
+            return expressionsInGroupBy.isPresent();
+        }
+
+        public boolean containsGroupBy(GroupingElement groupingElement)
+        {
+            for (Expression expression : groupingElement.getExpressions()) {
+                if (!containsGroupBy(expression)) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public boolean containsGroupBy(Expression expression)
```

Comment:
I think we need to provide better names to these APIs, they are doing lot more than simple contain check.